doc/GETTING_STARTED.md

```diff
@@ -0,0 +1,428 @@
+const WEBSOCKETS = require('libp2p-websockets')
+const SECIO = require('libp2p-secio')
+const MPLEX = require('libp2p-mplex')
+
+const node = await Libp2p.create({
+  modules: {
+    transport: [WEBSOCKETS],
+    connEncryption: [SECIO],
+    streamMuxer: [MPLEX]
+  }
+})
+```
+
+Protocols design is similar to an http api, where each protocol has an handler, receives a message and returns a response. If you want to create your own protocol, start by having a look at the [Protocol and Stream Muxing Example](../examples/protocol-and-stream-muxing).
+
+<details><summary>Read More</summary>
+If you want to know more about libp2p stream multiplexing, you should read the following content:
+
+- https://docs.libp2p.io/concepts/stream-multiplexing
+- https://github.com/libp2p/specs/tree/master/connections
+- https://github.com/libp2p/specs/tree/master/mplex
+</details>
+
+#### Running Libp2p
+
+Now that you have configured a [`Transport`][transport], [`Crypto`][crypto] and [**Stream Multiplexer**](streamMuxer) module, you can start your libp2p node. We can start and stop libp2p using the [`libp2p.start()`](./API.md#start) and [`libp2p.stop()`](./API.md#stop) methods.
+
+```js
+const Libp2p = require('libp2p')
+const WebSockets = require('libp2p-websockets')
+const SECIO = require('libp2p-secio')
+const MPLEX = require('libp2p-mplex')
+
+const node = await Libp2p.create({
+  modules: {
+    transport: [WebSockets],
+    connEncryption: [SECIO],
+    streamMuxer: [MPLEX]
+  }
+})
+
+// start libp2p
+await node.start()
+
+// stop libp2p
+await node.stop()
+```
+
+If you are not familiar with multiaddrs, you should read the [multiformats/js-multiaddr](https://github.com/multiformats/js-multiaddr) README.
+
+### Custom setup
+
+After having your libp2p node running, it is time to configure all its features according to your project needs.
+
+#### Peer Discovery
+
+In a P2P context, peer discovery is critical to a functioning system. It is not pratical to connect to all the peers we know manually, nor to get to know most of the peers' addresses in advance.
+
+Looking at the [available peer discovery](./CONFIGURATION.md#peer-discovery) protocols, there are several options to be considered:
+- If you already know the addresses of some other network peers, you should consider using `libp2p-bootstrap` as this is the easiest way of getting your peer into the network.
+- If it is likely that you will have other peers on your network, `libp2p-mdns` is a must. It allows peers to discover each other when on the same local network with zero configuration. mDNS uses a multicast system of DNS records.
+- In the event of working on a browser runtime, you must go with `js-libp2p-webrtc-star`.
+- if you need to crawl the network and discover a large number of peers, `js-libp2p-kad-dht`.
+
+Reviewing the events specification on [Libp2p.create](./API.md#create), each time a peer is discovered a `peer:discovery` event will be emitted by the node.
```

Comment:
Should we update the API doc so that we can deep link to the Events section? I think it convolutes this section and would make it easier of we could just link directly to that events declaration there.

Reply:
Should we add it in the same level as the Instance Methods and Types?

Reply:
Yeah, I think so